### PR TITLE
Item drops 3D!

### PR DESCRIPTION
**Pull request by Captainkittyca2**
Adds a toggle-able feature within `Enhancements->Graphics` to make Item drops 3D and spin. Most of the code is handled within a hook function in `3DItemDrops.cpp` file

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1559082527.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1559082983.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1559085617.zip)

### CODE REVIEW

**Comment by Caladius** [comment]
Using kafeis cpp file is a no no lol. You'll need to create your own cpp and h file following the other enhancement examples and self contain your code within it.

**Comment by Captainkittyca2** [comment]
> Using kafeis cpp file is a no no lol. You'll need to create your own cpp and h file following the other enhancement examples and self contain your code within it.

But It's within its own functions. I left the kafei functions untouched

**Comment by Caladius** [comment]
Each enhancement needs it's own separate file.

**Comment by Captainkittyca2** [comment]
> Each enhancement needs it's own separate file.

Alright, I'll make a h and cpp file for it

**Comment by Captainkittyca2** [comment]
Alright, added two files

**Review by garrettjoecox — COMMENTED** [review]
You should be able to pull this off without touching the actor’s file at all using the actor update & actor init hooks

**Comment by Captainkittyca2** [comment]
> You should be able to pull this off without touching the actor’s file at all using the actor update & actor init hooks

Ah I'm not familiar with those tbh.

**Comment by garrettjoecox** [comment]
Updated to use generic hooks & a few other tweaks

**Comment by Archez** [comment]
I've updated this PR with latest develop and applied some fixes/changes:
* Add rotation reset when disabling the enhancement so that bill-boarded sprites face the camera correctly.
* Refined which items should get the custom spin applied. Some where getting double rotation.
* Huge rupees weren't using the huge rupee model.
* Rupee scales are adjusted to match SoH.
* Removal of the rupee model gfx patch. This was impacting rupees from chests and in theory would not work with custom models.

If we wanted to render just the inner part of these rupees, we should setup new custom GID draw methods that just render the inner rupee part, but I don't think that is required for the scope of this PR.

**Review by Archez — APPROVED** [review]
LGTM 👍 Thanks to everyone who worked on this!
A good example of taking something from SoH, but porting it into hooks.

The changes for the slime actor in my opinion are fine as is, as we don't really have a good pattern for pulling something like that out.

File: mm/src/overlays/actors/ovl_En_Slime/z_en_slime.c [hunk 1]
```diff
@@ -1239,11 +1239,29 @@ void EnSlime_Draw(Actor* thisx, PlayState* play) {
...
+                    GetItem_Draw(play, GID_ARROWS_MEDIUM);
+                    break;
+            }
+        } else {
+            Matrix_Scale(0.03f, 0.03f, 0.03f, MTXMODE_APPLY);
+
+            gSPSegment(POLY_OPA_DISP++, 8, (uintptr_t)this->itemDropTex);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, gItemDropDL);
+        }
```
Comment (inspectredc) [inline mm/src/overlays/actors/ovl_En_Slime/z_en_slime.c]: Could this be VB hooked? smth like VB_DRAW_SLIME_ITEM(_DROP) ?

Comment (Archez) [reply mm/src/overlays/actors/ovl_En_Slime/z_en_slime.c]: Yeah you're right. I was only thinking about regular actor hooks. I've pushed an update with a VB hook register/unregister.

**Pull request merged**